content/docs/hooks-faq.md

```diff
@@ -18,116 +18,116 @@ This page answers some of the frequently asked questions about [Hooks](/docs/hoo
-  * [How do I implement shouldComponentUpdate?](#how-do-i-implement-shouldcomponentupdate)
-  * [How to memoize calculations?](#how-to-memoize-calculations)
-  * [How to create expensive objects lazily?](#how-to-create-expensive-objects-lazily)
-  * [Are Hooks slow because of creating functions in render?](#are-hooks-slow-because-of-creating-functions-in-render)
-  * [How to avoid passing callbacks down?](#how-to-avoid-passing-callbacks-down)
-  * [How to read an often-changing value from useCallback?](#how-to-read-an-often-changing-value-from-usecallback)
-* **[Under the Hood](#under-the-hood)**
-  * [How does React associate Hook calls with components?](#how-does-react-associate-hook-calls-with-components)
-  * [What is the prior art for Hooks?](#what-is-the-prior-art-for-hooks)
-
-## Adoption Strategy {#adoption-strategy}
-
-### Which versions of React include Hooks? {#which-versions-of-react-include-hooks}
-
-Starting with 16.8.0, React includes a stable implementation of React Hooks for:
+* **[एडॉप्शन स्ट्रेटेजी](#adoption-strategy)**
+  * [React के कौन से संस्करणों में hooks शामिल हैं?](#which-versions-of-react-include-hooks)
+  * [क्या मुझे अपने सभी क्लास कॉम्पोनेन्ट को फिर से लिखना होगा?](#do-i-need-to-rewrite-all-my-class-components)
+  * [मैं Hooks के साथ क्या कर सकता हूं जो मैं क्लासेज के साथ नहीं कर सकता हूं?](#what-can-i-do-with-hooks-that-i-couldnt-with-classes)
+  * [मेरा React ज्ञान कितना प्रासंगिक है?](#how-much-of-my-react-knowledge-stays-relevant)
+  * [क्या मुझे Hook, क्लासेज या दोनों का मिश्रण इस्तेमाल करना चाहिए?](#should-i-use-hooks-classes-or-a-mix-of-both)
+  * [क्या Hooks, क्लासेज के सारे उपयोगों को कवर करता है?](#do-hooks-cover-all-use-cases-for-classes)
+  * [क्या Hooks रेंडर प्रॉप्स और हायर आर्डर कंपोनेंट्स को रेप्लस करता है?](#do-hooks-replace-render-props-and-higher-order-components)
+  * [पॉपुलर APIs जैसे Redux connect() और React Router के लिए React के मायने?](#what-do-hooks-mean-for-popular-apis-like-redux-connect-and-react-router)
+  * [क्या Hooke स्टैटिक टाइपिंग के साथ काम करता है?](#do-hooks-work-with-static-typing)
+  * [Hooks का इस्तेमाल करते वाले कंपोनेंट्स का टेस्ट कैसे करें?](#how-to-test-components-that-use-hooks)
+  * [लिंट नियम क्या लागू करते हैं?](#what-exactly-do-the-lint-rules-enforce)
+* **[क्लासेज से हूक्स तक](#from-classes-to-hooks)**
+  * [लाइफ साइकिल मेथड्स Hooks को कैसे करेस्पॉन्ड करते हैं?](#how-do-lifecycle-methods-correspond-to-hooks)
+  * [Hooks के द्वारा डाटा फेचिंग कैसे करें?](#how-can-i-do-data-fetching-with-hooks)
+  * [क्या इंस्टैंस वेरिएबल की तरह कुछ है?](#is-there-something-like-instance-variables)
+  * [एक या एक से ज्यादा स्टेट वेरिएबल का इस्तेमाल करना चाहिए?](#should-i-use-one-or-many-state-variables)
+  * [क्या इफेक्ट को सिर्फ उपडटेस के बाद इस्तेमाल करना चाहिए?](#can-i-run-an-effect-only-on-updates)
+  * [पिछले प्रॉप्स या स्टेट को कैसे प्राप्त करें?](#how-to-get-the-previous-props-or-state)
```

Comment:
`प्रॉप्स` => `props`
`स्टेट` => `state`

we are not translating react keywords. update everywhere